src/NumericalAlgorithms/Spectral/Filtering.cpp

```diff
@@ -27,5 +28,98 @@ Matrix exponential_filter(const Mesh<1>& mesh, const double alpha,
   }
   return modal_to_nodal * filter_matrix * nodal_to_modal;
 }
+
+namespace {
+template <Spectral::Basis BasisType, Spectral::Quadrature QuadratureType>
+struct ZeroLowestModesImpl {
+  Matrix operator()(const size_t num_points,
+                    const size_t num_modes_to_zero) const noexcept {
+    const Matrix& nodal_to_modal =
+        Spectral::nodal_to_modal_matrix<BasisType, QuadratureType>(num_points);
+    const Matrix& modal_to_nodal =
+        Spectral::modal_to_nodal_matrix<BasisType, QuadratureType>(num_points);
+    Matrix filter_matrix(num_points, num_points, 0.0);
+    for (size_t i = num_modes_to_zero; i < num_points; ++i) {
+      filter_matrix(i, i) = 1.0;
+    }
+    return Matrix(modal_to_nodal * filter_matrix * nodal_to_modal);
+  }
+};
+}  // namespace
+
+const Matrix& zero_lowest_modes(const Mesh<1>& mesh,
+                                const size_t number_of_modes_to_zero) noexcept {
+  ASSERT(number_of_modes_to_zero <= mesh.extents(0),
+         "For a 1d mesh with " << mesh.extents(0)
+                               << " grid points you cannot zero "
+                               << number_of_modes_to_zero << '\n');
```

Comment:
I think you're missing a word here? `<< '\n');` -> `<< " modes.");`

I also suggest adding a comma after "grid points"